pkg/fedresolver/fedschema.go

```diff
@@ -0,0 +1,175 @@
+package fedresolver
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+// Define the httpClient interface
+type httpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// clientWrapper is a generic wrapper around http.Client or any httpClient implementation
+type clientWrapper struct {
+	client httpClient
+}
+
+func (c *clientWrapper) Do(req *http.Request) (*http.Response, error) {
+	return c.client.Do(req)
+}
+
+// Your SearchSchema struct with an interface for the HTTP client
+type SearchSchema struct {
+	routeToken []map[string]string
+	client     httpClient
+}
+
+type schemaPayload struct {
+	Data struct {
+		SearchSchema map[string][]string `json:"searchSchema"`
+	} `json:"data"`
+}
+
+func SearchSchemaResolver(ctx context.Context) (map[string]interface{}, error) {
+	routeToken := getTokenAndRoute()
+
+	// Create an http.Client
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+		Timeout: 30 * time.Second,
+	}
+
+	// Wrap the http.Client with the clientWrapper
+	wrappedClient := &clientWrapper{client: httpClient}
+
+	// Create a SearchSchema instance with the wrapped client
+	searchSchemaResult := &SearchSchema{
+		routeToken: routeToken,
+		client:     wrappedClient,
+	}
+
+	return searchSchemaResult.searchSchemaResults(ctx)
+}
+
+func (s *SearchSchema) searchSchemaResults(ctx context.Context) (map[string]interface{}, error) {
+	klog.Info("Resolving searchSchemaResults()")
+	srchSchema := map[string]interface{}{}
+	var schemaErr error
+	propArray := []interface{}{}
+	if schemaErr == nil {
+		klog.Info("schemaErr is nill here")
+	}
+	// These default properties are always present and we want them at the top.
+	defaultschema := []string{"cluster", "kind", "label", "name", "namespace", "status"}
+	// Use a map to remove duplicates efficiently.
+	defaultschemaMap := map[string]struct{}{}
+	newschemaMap := map[string]struct{}{}
+	for _, key := range defaultschema {
+		defaultschemaMap[key] = struct{}{}
+		propArray = append(propArray, key)
+	}
+	// Return the default properties if there is error in processing the rest
+	srchSchema["allProperties"] = propArray
+
+	if len(s.routeToken) == 0 {
+		klog.Warning("No routes and tokens provided.")
+		return srchSchema, fmt.Errorf("error in search schema call: No leaf hub routes and tokens provided")
+	}
+
+	var wg sync.WaitGroup
+	for _, routeAndToken := range s.routeToken {
+		wg.Add(1)
+		routeAndToken := routeAndToken
+		// Wrap the worker call in a closure that makes sure to tell
+		// the WaitGroup that this worker is done.
+		go func() {
```

Comment:
Maybe it's better to pass as a parameter to the go func.

```suggestion
		// Wrap the worker call in a closure that makes sure to tell
		// the WaitGroup that this worker is done.
		go func(routeAndToken string) {
		  //...
		}(routeAndToken)
```